pyproject.toml

```diff
@@ -45,3 +72,87 @@ ignore_missing_imports = true
 [[tool.mypy.overrides]]
 module = "click_log"
 ignore_missing_imports = true
+
+[tool.ruff]
+line-length = 100
+
+[tool.ruff.lint]
+extend-select = ["I"]
+
+[tool.dev-cmd.commands]
+fmt = ["ruff", "format"]
+check-fmt = ["ruff", "format", "--diff"]
+
+lint = ["ruff", "check", "--fix"]
+check-lint = ["ruff", "check"]
+
+type-check = ["mypy", "science", "tests", "test-support", "scripts"]
+
+doc = ["sphinx-build", "-b", "html", "-aEW", "docs", "docs/build/html"]
+linkcheck = ["sphinx-build", "-b", "linkcheck", "-aEW", "docs", "docs/build/linkcheck"]
+
+create-zipapp = ["python", "scripts/create-zipapp.py"]
+
+[tool.dev-cmd.commands.package-thin-scie]
+env = {"SCIENCE_LIFT_BUILD_DEST_DIR" = "dist"}
+args = [
+    "python",
+    "dist/science.pyz",
+    "lift",
+    "--file",
+    "science.pyz=dist/science.pyz",
+    "--file",
+    "docsite=docs/build/html",
+    "--include-provenance",
+    "build",
+    "--hash",
+    "sha256",
+    "--use-platform-suffix",
+]
+
+[tool.dev-cmd.commands.package-fat-scie]
+env = {"SCIENCE_LIFT_BUILD_DEST_DIR" = "dist"}
+args = [
+    "python",
+    "dist/science.pyz",
+    "lift",
+    "--file",
+    "science.pyz=dist/science.pyz",
+    "--file",
+    "docsite=docs/build/html",
+    "--include-provenance",
+    "--invert-lazy",
+    "cpython",
+    "--app-name",
+    "science-fat",
+    "build",
+    "--hash",
+    "sha256",
+    "--use-platform-suffix",
+]
+
+[tool.dev-cmd.commands.run-zipapp]
+env = {"SCIENCE_DOC_LOCAL" = "docs/build/html"}
+args = ["dist/science.pyz"]
+accepts-extra-args = true
+
+[tool.dev-cmd.commands.pytest]
+args = ["pytest", "-n", "auto"]
+cwd = "tests"
+accepts-extra-args = true
+[tool.dev-cmd.commands.pytest.env]
+BUILD_ROOT = ".."
+PYTHONPATH = "../test-support"
+SCIENCE_TEST_PYZ_PATH = "../dist/science.pyz"
+
+[tool.dev-cmd.tasks]
+test = ["create-zipapp", "pytest"]
+checks = ["fmt", "lint", "type-check", "test"]
+ci = [["check-fmt", "check-lint", "type-check", "test"]]
+run = [["doc", "create-zipapp"], "run-zipapp"]
+
+package = [["doc", "create-zipapp"], ["package-thin-scie", "package-fat-scie"]]
```

Comment:
The nested lists run in parallel. So `doc` & `create-zipapp` run in parallel, then, when both are complete, `package-thin-scie` & `package-fat-scie` run in parallel. The serial / parallel / serial / ... nesting switches can be carried on as deeply nested as desired.